src/frame/reader.rs

```diff
@@ -53,7 +53,10 @@ impl<R: BlockRead + Unpin> FrameReader<R> {
         if !need_to_skip_block {
             return Ok(());
         }
-        self.reader.next_block().await?;
+        if !self.reader.next_block().await? {
+            return Err(ReadFrameError::NotAvailable);
+        }
```

Comment:
this fixes a bug where the FrameReader would sometime loop over the last block indefinitely